File: backend/queute/src/main/java/tn/cloudnine/queute/model/roadmap/RoadMap.java
package tn.cloudnine.queute.model.roadmap;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.CascadeType;
import tn.cloudnine.queute.model.user.User;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RoadMap {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id ;
    private String title ;
    @Column(length = 1000)
    private String description ;
    private int nbrApprove ;
    private int nbrDisapproval;
    @OneToMany
    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    private List<RoadMapApproval> approvals ;

    @ManyToOne(cascade = jakarta.persistence.CascadeType.PERSIST)
    private User creator ;
    @OneToMany
    @Cascade(CascadeType.ALL)
    private List<Step> steps ;

    @Override
    public String toString() {
        return "RoadMap{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", nbrApprove=" + nbrApprove +
                ", nbrDisapproval=" + nbrDisapproval +
                ", approvals=" + approvals +
                ", creator=" + creator +
                ", steps=" + steps +
                '}';
    }
}

File: backend/template/src/app/models/RoadMap.ts
import { User } from './User';
import { Step } from './Step';


        export class RoadMap {
                id: number|null;
                title: string;
                description: string;
                creator: User;
                nbrApprove:number;
                nbrDisapproval:number;
                steps: Step[]=[];
            
                constructor(init?: Partial<RoadMap>) {
                        this.id = init?.id ?? null;
                        this.title=init?.title ??"" ; 
                        this.description=init?.description ?? "";
                        this.nbrApprove=init?.nbrApprove ?? 0;
                        this.nbrDisapproval=init?.nbrDisapproval ?? 0;
                        this.creator=init?.creator ?? new User() ;
                        this.steps = (init?.steps ?? []).map(step => new Step(step));
                    }
            }